lib/rules/valid-jsdoc.js

```diff
@@ -137,7 +137,7 @@ module.exports = function(context) {
         var elements = [];
 
         if (tag.type.type === "TypeApplication") { // {Array.<String>}
-            elements = tag.type.applications;
+            elements = tag.type.applications[0].type === "UnionType" ? tag.type.applications[0].elements : tag.type.applications;
```

Comment:
I thought this required a larger refactor to handle nested structures but looks like this simple fix does it :)

Nice job!